### ISSUE TITLE
Avoid find_characteristics_by_aid_iid making bluez dbus calls

### ISSUE THREAD

**Issue by Jc2k** [jlusiardi/homekit_python#73]
I think we can avoid scraping the characteristics descriptors over dbus entirely if the cached pairing data is up to date? If we can maintain the iid -> uuid mappings in our JSON cache then we only have to loop over the services and characteristics and compare uuids, we dont have to read the descriptors again.

This would kind of replace the _char_by_aid_iid cache.

**Comment by Jc2k** [comment]
The idea is to replace something like:

```
def find_characteristic_by_iid(device, iid):
    """

    :param device:
    :param iid: instance id
    :return:
    """
    service_found = None
    for service in device.services:
        logger.debug('Searching service: %s', service.uuid)
        for characteristic in service.characteristics:
            logger.debug('Searching characteristic: %s', characteristic.uuid)
            for descriptor in characteristic.descriptors:
                if descriptor.uuid == CharacteristicInstanceID:
                    logging.debug('Reading CharacteristicInstanceID')
                    value = descriptor.read_value()
                    cid = int.from_bytes(value, byteorder='little')
                    if iid == cid:
                        logging.debug('iid %d -> uuid %s', iid, characteristic.uuid)
                        return (characteristic, cid)

    logging.error('Service with instance id %d not found', iid)
    return None, None
```

with

```
def find_characteristic_by_iid(device, iid):
    for service in pairing_data['accessories'][0]['services']:
        for char in service['characteristics']:
            if char['iid'] != iid:
                continue

            for s in device.services:
                if s.uuid != service['uuid']:
                    continue
                for c in s.characteristics:
                    if c.uuid == char['uuid']:
                        return c
```

(pseudo code off top of head, not intended to be functional)

Essentially this is about mapping (and then caching) characteristic instance ids to characteristic uuids. The UUIDs are cheaper, as a comparison against the CharacteristicInstanceID requires a `descriptor.read_value` which is a dbus RPC round trip, where as the uuid is already populated.

Right now we do cache this relationship at runtime (iid -> characteristic object) but I think it can be stored in the JSON and reused between processes.

**Comment by Jc2k** [comment]
This can be closed now as #76 merged.
